shared-libs/registration-utils/src/index.js

```diff
@@ -0,0 +1,28 @@
+// Returns whether `doc` is a valid registration against a configuration
+// This is done by checks roughly similar to the `registration` transition filter function
+// Serves as a replacement for checking for `transitions` metadata within the doc itself
+exports.isValidRegistration = function(doc, settings) {
+  if (!doc ||
+      (doc.errors && doc.errors.length) ||
+      !settings ||
+      !settings.registrations ||
+      doc.type !== 'data_record' ||
+      !doc.form) {
+    return false;
+  }
+
+  var registrationConfiguration = settings.registrations.find(function(conf) {
+    return new RegExp('^\W*' + conf.form + '\\W*$','i').test(doc.form);
```

Comment:
I know this is how the old code was but it seems really wasteful to create a regex for each configured registration every time this is called. Instead can we trim and lowercase the doc.form and just compare with the configured string?